wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
This is not a good idea.
Two threads should not work on the same page instance at the same time.
This will lead to concurrency related issues.
E.g. two Ajax requests will have their own Request Cycles but might work on the same page instance this way. Without AsyncContext being needed at all.